### PR TITLE
Add loadMol2 and loadMol3 support to AmberParameterSet.from_leaprc

### LINKED ISSUE #592: Does from_leaprc() support mol2 files?
I tried to use the following `leaprc`:
```
naproxen = loadmol2 molecule-0000000000.gaff.mol2
loadamberparams molecule-0000000000.frcmod
aspirin = loadmol2 molecule-0000000001.gaff.mol2
loadamberparams molecule-0000000001.frcmod
imatinib = loadmol2 molecule-0000000002.gaff.mol2
loadamberparams molecule-0000000002.frcmod
bosutinib = loadmol2 molecule-0000000003.gaff.mol2
loadamberparams molecule-0000000003.frcmod
```
to generate an `ffxml` file via
```python
    leaprc = StringIO(leaprc)
    params = parmed.amber.AmberParameterSet.from_leaprc(leaprc)
    params = parmed.openmm.OpenMMParameterSet.from_parameterset(params)
    ffxml = StringIO()
    params.write(ffxml)
```
but ended up with just the `frcmod` parameters but no residue templates:
```
<ForceField>
 <Info>
  <DateGenerated>2016-02-04</DateGenerated>
 </Info>
 <AtomTypes>
 </AtomTypes>
 <Residues>
 </Residues>
 <PeriodicTorsionForce>
  <Proper type1="ca" type2="cp" type3="nb" type4="ca" periodicity1="2" phase1="3.14159265359" k1="20.0832"/>
  <Proper type1="cp" type2="cp" type3="nb" type4="ca" periodicity1="2" phase1="3.14159265359" k1="20.0832"/>
  <Proper type1="ca" type2="ca" type3="cg" type4="n1" periodicity1="2" phase1="3.14159265359" k1="0.0"/>
  <Improper type1="ca" type2="ca" type3="ca" type4="ha" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="ca" type2="ca" type3="ca" type4="ca" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="ca" type2="ca" type3="ca" type4="os" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="ca" type2="c" type3="ca" type4="ca" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="c" type2="ca" type3="o" type4="oh" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="c" type2="c3" type3="o" type4="os" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="ca" type2="ca" type3="cp" type4="ha" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="ca" type2="ca" type3="h4" type4="nb" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="ca" type2="cp" type3="h4" type4="nb" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="cp" type2="ca" type3="ca" type4="cp" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="ca" type2="ca" type3="ca" type4="n" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="ca" type2="ca" type3="ca" type4="nh" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="cp" type2="ca" type3="cp" type4="nb" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="ca" type2="nb" type3="nb" type4="nh" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="c" type2="ca" type3="n" type4="o" periodicity1="2" phase1="3.14159265359" k1="43.932"/>
  <Improper type1="nh" type2="ca" type3="ca" type4="hn" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="n" type2="c" type3="ca" type4="hn" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="ca" type2="ca" type3="ca" type4="cg" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
  <Improper type1="ca" type2="ca" type3="ca" type4="nb" periodicity1="2" phase1="3.14159265359" k1="4.6024"/>
 </PeriodicTorsionForce>
</ForceField>
```
I'm happy to take a stab at implementing what would be needed for molecules loaded by `loadmol2` to also appear as residue templates---just let me know what I would need to do.

### PR DESCRIPTION
**Pull request by swails**
Closes #592 

@rafwiewiora seems to have his hands full with FF validation, so I went ahead and handled this.  Still need to implement an `AmberParameterSet.join` function in order to support "source", but that's for another PR.

### CODE REVIEW

**Comment by rafwiewiora** [comment]
Thanks a lot @swails!

**Comment by jchodera** [comment]
Awesome, thanks!  Needed this for https://github.com/choderalab/openmoltools/pull/190

**Comment by rafwiewiora** [comment]
Yeah, looks from the complexity of this that it would have been faster for you to do this than me anyway, even if I wasn't busy.

File: parmed/amber/parameters.py [hunk 1]
```diff
@@ -264,23 +269,52 @@ def from_leaprc(cls, fname, search_oldff=False):
...
+            if _loadparamsre.findall(line):
+                fname = process_fname(_loadparamsre.findall(line)[0])
+                params.load_parameters(_find_amber_file(fname, search_oldff))
+            elif _loadoffre.findall(line):
+                fname = process_fname(_loadoffre.findall(line)[0])
+                params.residues.update(
                     AmberOFFLibrary.parse(_find_amber_file(fname, search_oldff))
-            )
+                )
+            elif _loadmol2re.findall(line):
```
Comment (hainm) [inline parmed/amber/parameters.py]: it's likely that if not having `else` will reduce the coverage score. (you only have if ... elif)

Comment (swails) [inline parmed/amber/parameters.py]: Why would it have any affect on the coverage score?  `else: pass` is a no-op.  Even if I turned on branch coverage that branch would still be triggered.

Also, this logic shows the intent.  Each "if" handles a case that is processed from the leaprc file and everything else is ignored (like "loadPDB" and "solvateOct/Box").  As you support more commands (like `source` is on the docket next), then you just add another "elif" block.

Comment (hainm) [inline parmed/amber/parameters.py]: > Why would it have any affect on the coverage score

I am not sure. :D I got this for pytraj. Can you turn on code coverage highlight on coveralls so we can look at that? (like this https://coveralls.io/files/2034459165)

**Pull request merged**